docs/platform/tutorials/eventDetection/crashDetection.md

```diff
@@ -0,0 +1,277 @@
+
+10. Store the `loaded_model` in a class property by adding the following line to the the `__init__` function
+
+	```py
+	self.loaded_model = loaded_model
+	```
+
+9. Now replace the current `on_pandas_frame_handler` with this code:
+
+	```py
+		def on_pandas_frame_handler(self, df: pd.DataFrame):
+			
+			if "gForceX" in df:
+				df["gForceTotal"] = df["gForceX"].abs() +  df["gForceY"].abs() + df["gForceZ"].abs()
+				df["shaking"] = self.loaded_model.predict(df[["gForceZ", "gForceY", "gForceX", "gForceTotal"]])
+
+				if df["shaking"].max() == 1: 
+					print("Crash detected.")
+
+					self.output_stream.events.add_timestamp_nanoseconds(df.iloc[0]["time"]) \
+						.add_value("crash", "Crash detected.") \
+						.write()
+	```
+
+	This code first checks that the required data is in the dataframe then uses the ML model with the g-force data to determine when shaking is occurring.
+
+	If shaking is occurring an event is generated and streamed to the output topic.
+
+	???- info "The completed `quix_function.py` should look like this"
+
+		```py
+		from quixstreaming import StreamReader, StreamWriter, EventData, ParameterData
+		import pandas as pd
+
+		class QuixFunction:
+			def __init__(self, input_stream: StreamReader, output_stream: StreamWriter, loaded_model):
+				self.input_stream = input_stream
+				self.output_stream = output_stream
+				self.loaded_model = loaded_model
+
+			# Callback triggered for each new event.
+			def on_event_data_handler(self, data: EventData):
+				print(data.value)
+				# Here transform your data.
+				self.output_stream.events.write(data)
+
+			# Callback triggered for each new parameter data.
+			def on_pandas_frame_handler(self, df: pd.DataFrame):
+				
+				# if the expected column is in the dataframe
+				if "gForceX" in df:
+					# calc total g-force
+					df["gForceTotal"] = df["gForceX"].abs() +  df["gForceY"].abs() + df["gForceZ"].abs()
+					# predict 'shaking'
+					df["shaking"] = self.loaded_model.predict(df[["gForceZ", "gForceY", "gForceX", "gForceTotal"]])
+
+					# if 'shaking'
+					if df["shaking"].max() == 1: 
+						print("Crash detected.")
+
+						# write the event to the output stream
+						self.output_stream.events.add_timestamp_nanoseconds(df.iloc[0]["time"]) \
+							.add_value("crash", "Crash detected.") \
+							.write()
+		```
+	
+### Test again
+
+You can once again run the code in the development environment to test the functionality.
+
+1. Ensure that the data source you deployed earlier is running
+
+2. Click `Run` in the top right of the browser to run the event detection code
+
+3. If you chose to stream live data then gently shake your phone 
+
+4. If you went for the CSV data then wait for a crash event to be streamed from the data set or stop and start the service in another browser tab
+
+5. Observe the `Console` tab. You should see a message saying "Crash detected"
+
+4. On the `Messages` tab select `output : phone-out` from the first drop down
```

Comment:
> from the first drop down

Drop-down as per Microsoft and also [here](https://en.wikipedia.org/wiki/Drop-down_list). Also, other instances below.